srml/contract/src/wasm/prepare.rs

```diff
@@ -29,20 +29,34 @@ use rstd::prelude::*;
 use runtime_primitives::traits::As;
 
 struct ContractModule<'a, Gas: 'a> {
-	// An `Option` is used here for loaning (`take()`-ing) the module.
-	// Invariant: Can't be `None` (i.e. on enter and on exit from the function
-	// the value *must* be `Some`).
+	/// A deserialized module. The module is valid (this is Guaranteed by `new` method).
+	///
+	/// An `Option` is used here for loaning (`take()`-ing) the module.
+	/// Invariant: Can't be `None` (i.e. on enter and on exit from the function
```

Comment:
This sounds like a job for a function that takes a callback.  Something like

```rust
fn with_module<U, T: FnOnce(elements::Module) -> (elements::Module, U)>(cb: T) -> U {
	let (new_module, retval) = cb(self.module.take());
	self.module = Some(new_module);
	retval
}
```

This would enforce that `ContractModule::module` is never `None` unless something `panic!`d.